### PR TITLE
Make http3 codec not encode empty trailers block

### PR DESCRIPTION
**Pull request by ravenblackx**
Commit Message: Make http3 codec not encode empty trailers block
Additional Description: http3 and `grpc_web_filter` combined resulted in an incompatibility with Chrome - a grpc web request received a response with an empty trailers block, which Chrome treats as a network error. The http2 codec had a special case in it for empty trailers, which was not mimicked in the http3 codec. This PR makes the http3 codec similar to the http2 codec in this respect, and adds an integration test ensuring that this behavior is in all downstream codecs.
Risk Level: Small. It is a behavior change, but from an already malfunctioning state in a rare edge case.
Testing: Added integration test which fails before the change and passes after.
Docs Changes: n/a
Release Notes: Yes
Platform Specific Features: n/a

### CODE REVIEW

**Comment by ravenblackx** [comment]
(Failure appears to be a flake added in #34461)

**Review by RyanTheOptimist — COMMENTED** [review]
Nice! Thanks for fixing this. I think we should probably have a runtime guard for this, on the off chance that this behavior change breaks someone. LGTM otherwise.

File: changelogs/current.yaml [hunk 1]
```diff
@@ -107,7 +107,9 @@ bug_fixes:
 - area: http3
   change: |
     Fixed a bug where an empty trailers block could be sent. This would occur if a filter removed
-    the last trailer - a likely occurrence with the ``grpc_web_filter``.
+    the last trailer - a likely occurrence with the ``grpc_web_filter``. This change makes http3 codec
+    behave the same way http2 codec does, converting an empty trailers block to no trailers.
```
Comment (phlax) [inline changelogs/current.yaml:111]:
```suggestion
    behave the same way HTTP/2 codec does, converting an empty trailers block to no trailers.
```

File: changelogs/current.yaml [hunk 2]
```diff
@@ -107,7 +107,9 @@ bug_fixes:
 - area: http3
   change: |
     Fixed a bug where an empty trailers block could be sent. This would occur if a filter removed
-    the last trailer - a likely occurrence with the ``grpc_web_filter``.
+    the last trailer - a likely occurrence with the ``grpc_web_filter``. This change makes http3 codec
```
Comment (phlax) [inline changelogs/current.yaml:110]:
```suggestion
    the last trailer - a likely occurrence with the ``grpc_web_filter``. This change makes HTTP/3 codec
```

**Review by RyanTheOptimist — APPROVED** [review]

File: changelogs/current.yaml [hunk 3]
```diff
@@ -107,7 +107,9 @@ bug_fixes:
...
+    behave the same way http2 codec does, converting an empty trailers block to no trailers.
+    This behavior can be reverted by setting the runtime guard ``envoy.reloadable_features.http3_remove_empty_trailers`` to false.
```
Comment (phlax) [inline changelogs/current.yaml:112]:
```suggestion
    This behavior can be reverted by setting the runtime guard ``envoy.reloadable_features.http3_remove_empty_trailers`` to ``false``.
```

**Review by phlax — COMMENTED** [review]
docs/changelog nits ...

**Review by phlax — APPROVED** [review]
lgtm, thanks!

**Pull request merged**